backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/CalendarsService.cs

```diff
@@ -217,35 +217,56 @@ public async Task<bool> SyncChangesFromGoogleCalendar(string email)
 
             foreach (var calendar in calendars)
             {
-                var visibleCalendar = await _context.CalendarVisibleForTeams.Where(x => x.CalendarId == calendar.Id).ToListAsync();
-
-                await RemoveCalendarMeetings(visibleCalendar, calendar.Id);
-                await AddMeetingsFromCalendar(email, calendar.VisibleForTeams, calendar.Id);
+                await RemoveCalendarMeetings(calendar.Id);
+                await AddMeetingsFromCalendar(email, calendar.Id);
             }
 
             await _context.SaveChangesAsync();
 
             return true;
         }
 
-        private async Task RemoveCalendarMeetings(IEnumerable<CalendarVisibleForTeam> visibleCalendar, long calendarId)
+        private async Task RemoveCalendarMeetings(long calendarId)
         {
-            foreach (var item in visibleCalendar.ToList())
+            await _calendarEventService.RemoveCalendarEvents(calendarId);
+        }
+
+        public async Task CancelMeetingInGoogleCalendar(string meetingName, Calendar calendar)
+        {
+            var events = await GetEventsFromGoogleCalendar(calendar.ConnectedCalendar);
+            var cancelledEvent = events.FirstOrDefault(el => el.Summary == meetingName);
+
+            if (cancelledEvent is null)
             {
-                await _calendarEventService.RemoveCalendarEvents(calendarId);
+                return;
             }
+            
+            var body = new
+            {
+                status = "cancelled",
+                end = cancelledEvent.End,
+                start = cancelledEvent.Start,
```

Comment:
why we need to update end and start time if we just need to change status for cancelled?

Reply:
Google's official documentation says that these options are required. In other words, the request will not be accepted without these parameters